src/main/scala/kamon/logback/instrumentation/AsyncAppenderInstrumentation.scala

```diff
@@ -78,19 +82,27 @@ class AsyncAppenderInstrumentation {
     val span = currentContext.get(Span.ContextKey)
 
     if (span.context().traceID != IdentityProvider.NoIdentifier && AsyncAppenderInstrumentation.mdcContextPropagation){
-      AsyncAppenderInstrumentation.mdcTracedKeys.foreach { tracedkey =>
-        currentContext.get(Key.broadcastString(tracedkey)).foreach { tracedKeyValue =>
-          MDC.put(tracedkey, tracedKeyValue)
-        }
-      }
       MDC.put(AsyncAppenderInstrumentation.mdcTraceKey, span.context().traceID.string)
       MDC.put(AsyncAppenderInstrumentation.mdcSpanKey, span.context().spanID.string)
+
+      AsyncAppenderInstrumentation.mdcLocalKeys.foreach { localKey =>
+        currentContext.get(localKey).foreach { localKeyValue =>
+          MDC.put(localKey.name, localKeyValue)
+        }
+      }
+
+      AsyncAppenderInstrumentation.mdcBroadcastKeys.foreach { broadcast =>
+        currentContext.get(broadcast).foreach { broadcastKeyValue =>
+          MDC.put(broadcast.name, broadcastKeyValue)
```

Comment:
I think that in Kamon you can have 2 keys with the same name, one local, one broadcast. Because the map that keeps them is of type `Map[Key[_], Any]` so it can distinguish between the two. But in MDC, we only have the key names, so they might conflict... 🤔 

Maybe, for now, we could just add a note in the README about this...